src/main/java/servlets/UploadReceiptServlet.java

```diff
@@ -53,8 +54,33 @@
   private static final String DEV_SERVER_BASE_URL = "http://0.0.0.0:80";
   // Matches JPEG image filenames.
   private static final Pattern validFilename = Pattern.compile("([^\\s]+(\\.(?i)(jpe?g))$)");
-  // Logs to System.err by default.
-  private static final Logger logger = Logger.getLogger(UploadReceiptServlet.class.getName());
+
+  private static Logger logger;
+  private static BlobstoreService blobstoreService;
+  private static BlobInfoFactory blobInfoFactory;
+  private static DatastoreService datastore;
+  private static Clock clock;
```

Comment:
I would make these final but not static. I would also be careful about making instance variables static (see http://www.object-oriented-security.org/lets-argue/singletons).